tensorrt_llm/_torch/pyexecutor/decoder.py

```diff
@@ -20,40 +20,53 @@
                                                     SpeculativeDecodingMode)
 from tensorrt_llm.mapping import Mapping
 
-from .llm_request import LlmRequest, LlmRequestState
+from .llm_request import LlmRequest, LlmRequestState, LogProbs
 from .scheduler import ScheduledRequests
 
 
+@dataclass
+class DecoderState:
+    scheduled_requests: ScheduledRequests
```

Comment:
I notice that we introduce `DecoderState` and `BatchDecoder` in this PR. It makes sense to me to use a data class to encapsulate some states. 

Just for better code review, could you please split this PR to some small PRs:

1. a PR for introducing `DecoderState` and `BatchDecoder` only
2. a PR for support return logits based on PR 1.

Reply:
Created !3234.